packages/xo-server/docs/vm-sync-hook.md

```diff
@@ -0,0 +1,80 @@
+# VM Sync Hook
+
+> This feature is currently _unstable_ and might change or be removed in the future.
+>
+> Feedbacks are very welcome on the [project bugtracker](https://github.com/vatesfr/xen-orchestra/issues).
+
+Before snapshotting (with or without memory, ie checkpoint), XO can notify the VM via an HTTP request.
+
+A typical use case is to make sure the VM is in a consistent state during the snapshot process, for instance by making sure database writes are flushed to the disk.
+
+## Configuration
+
+The feature is opt-in via a tag on the VM: `xo:notify-on-snapshot`.
+
+By default, it will be an HTTP request on the port `1727`, on the first IP address reported by the VM.
+
+If the _VM Tools_ (i.e. management agent) are not installed on the VM or if you which to use another URL, you can specify this in the tag: `xo:notify-on-snapshot=<URL>`.
+
+## Specification
+
+XO will waits for the request to be answered before starting the snapshot, but will not wait longer than _1 minute_.
+
+If the request fails for any reason, XO will go ahead with snapshot immediately.
+
```

Comment:
should we add a shared secret to ensure the query comes from xo ?

Reply:
I thought about it and while I think it's a good idea, I believe it can wait for a second version.

Also, the URL is configurable and users can use both HTTPS and add credentials if necessary.

Reply:
but url configuration means that we lose the auto ip discovery ?

Reply:
Unfortunately yes.

I'm open to suggestions if you have ideas, but I'd like to keep the first iteration simple.

Reply:
Adding a secret in the config is easy but unfortunately I don't know how to share with with XO Proxy.

I think this adds complexity, which will confuse users and we should expect some support ticket regarding this.

But that may be fine for a first version.